demos/sample_azure_iot_ca_recovery/sample_azure_iot_ca_recovery.c

```diff
@@ -334,6 +334,39 @@ static uint32_t prvSetupRecoveryNetworkCredentials( NetworkCredentials_t * pxNet
 }
 /*-----------------------------------------------------------*/
 
+static void runRecovery( NetworkCredentials_t * xNetworkCredentials,
+                         uint8_t ** pucIotHubHostname,
+                         uint32_t * pulIothubHostnameLength,
+                         uint8_t ** pucIotHubDeviceId,
+                         uint32_t * pulIothubDeviceIdLength )
+{
+    LogInfo( ( "In recovery\r\n" ) );
+
+    memset( xNetworkCredentials, 0, sizeof( *xNetworkCredentials ) );
+    uint32_t ulStatus = prvSetupRecoveryNetworkCredentials( xNetworkCredentials );
+    configASSERT( ulStatus == 0 );
+
+    if( ( ulStatus = prvRunRecovery( xNetworkCredentials ) ) != 0 )
+    {
+        LogError( ( "Failed to run recovery error code = 0x%08x\r\n", ulStatus ) );
+        configASSERT( ulStatus == 0 );
+    }
+    else if( ( ulStatus = prvSetupNetworkCredentials( xNetworkCredentials ) ) != 0 )
+    {
+        LogError( ( "Could not set network credentials\r\n" ) );
+        configASSERT( ulStatus == 0 );
+    }
+    else if( ( ulStatus = prvIoTHubInfoGet( xNetworkCredentials, pucIotHubHostname,
+                                            pulIothubHostnameLength, pucIotHubDeviceId,
+                                            pulIothubDeviceIdLength ) ) != 0 )
+    {
+        LogError( ( "Failed to run DPS after recovery!: error code = 0x%08x\r\n", ulStatus ) );
+        configASSERT( ulStatus == 0 );
```

Comment:
These asserts are guaranteed to fail if they are evaluated - should we have configAssert( false ) to be clearer about the flow?